shared/js/https.js

```diff
@@ -1,228 +1,211 @@
-    load.JSONfromLocalFile(constants.httpsWhitelist, (wl) => knownMixedContentList = wl)
-})
+// number of chunks to split up the data
+// when storing to Safari to avoid over quota issues:
+const LOCAL_STORAGE_CHUNKS = 5
+
+let httpsUpgradeList = []
 
 class HTTPS {
 
     constructor () {
-        this.db = null
-        this.dbObjectStore = 'https'
-
-        // if this.isSync: load Chrome-only synchronous rules object into memory
-        this.isSync = utils.isChromeBrowser() ? true : false
-        this.syncRuleCache = {}
-
         this.isReady = false
-        this._ready = this.init().then(() => this.isReady = true)
+        this.init().then(() => this.isReady = true)
+
+        // in the background, try to update the list:
+        this.updateList()
 
         return this
     }
 
     init () {
-        return new Promise((resolve) => {
-            db.ready().then(() => {
-                this.db = db
+        console.log("HTTPS: init()")
 
-                if (this.isSync) {
-                    this.initSync().then(() => resolve())
-                } else {
-                    resolve()
-                }
+        return new Promise((resolve) => {
+            // Try to load the list from local storage
+            this.getFromStorage().then((items) => {
+
+                // if items were found in local storage,
+                console.log("HTTPS: init() found existing list in storage with " + items.length + " items")
+                httpsUpgradeList = items
+                return resolve()
+
+            }, () => {
+
+                // if it fails to find anything in local storage,
+                // load from the json file packaged with the extension
+                console.log("HTTPS: init() nothing found in storage, loading from local file")
+
+                settings.ready().then(() => {
+                    load.JSONfromLocalFile(constants.httpsUpgradeList, (items) => {
+                        console.log("HTTPS: init() loaded upgrade list from local file with " + items.length + " items")
+                        httpsUpgradeList = items
+                        return resolve()
+                    })
+                })
             })
         })
     }
 
-    initSync () {
-        return new Promise((resolve) => {
-            this.db.logAllRecords(this.dbObjectStore, this.syncRuleCache)
-            resolve()
+    updateList() {
+        console.log("HTTPS: updateList() check if new list exists")
+
+        let etag = settings.getSetting('https-etag') || ''
+        
+        load.loadExtensionFile({
+            url: constants.httpsUpgradeListUrl,
+            source: 'external',
+            etag: etag
+        }, (data, res) => {
+            // this only gets called if the etag is different
+            // and it was able to get a new list
+            console.log("HTTPS: updateList() got updated list from server")
+
+            let newEtag = res.getResponseHeader('etag') || ''
```

Comment:
Does `load.loadExtensionFile` handle the eTag? Namely, where does the tag get checked / skipped everything? Is it a nop from loadExtensionFile if the tag doesn't pass muster?

Reply:
yes, the comment a few lines above was trying to say that, will make it clearer